File: generate_observables.cpp
#include <cstdio>
using namespace std;

int system_size = 20;

int main(){
    freopen("generated_observables.txt", "w", stdout);
    printf("%d\n", system_size);

    for(int i = 0; i < system_size - 1; i++){
        for(int j = 0; j < system_size - 1; j++){
            if(j == i || j == i + 1 || j+1 == i) continue;
            printf("4 Y %d Y %d X %d X %d\n", i, i+1, j, j+1);
        }
    }
    for(int i = 0; i < system_size - 1; i++){
        for(int j = 0; j < system_size; j++){
            if(j == i || j == i + 1) continue;
            for(int j2 = 0; j2 < system_size; j2++){
                if(j2 == i || j2 == i + 1 || j2 == j) continue;
                printf("4 X %d X %d Z %d Z %d\n", i, i+1, j, j2);
            }
        }
    }
    for(int i = 0; i < system_size - 1; i++){
        for(int j = 0; j < system_size; j++){
            if(j == i || j == i + 1) continue;
            printf("3 X %d X %d Z %d\n", i, i+1, j);
        }
    }
}

File: generate_observables.py
observable_file = open('generated_observables.txt', 'w')

system_size = 20
print(system_size, file = observable_file)

for i in range(system_size - 1):
    for j in range(system_size - 1):
        if j == i or j == i + 1 or j+1 == i: continue
        print("4 Y {} Y {} X {} X {}".format(i, i+1, j, j+1), file = observable_file)

for i in range(system_size - 1):
    for j in range(system_size):
        if j == i or j == i + 1: continue
        for j2 in range(system_size):
            if j2 == i or j2 == i + 1 or j2 == j: continue
            print("4 X {} X {} Z {} Z {}".format(i, i+1, j, j2), file = observable_file)

for i in range(system_size - 1):
    for j in range(system_size):
        if j == i or j == i + 1: continue
        print("3 X {} X {} Z {}".format(i, i+1, j), file = observable_file)
